packages/core/src/components/ui/ProductShelf/ProductShelf.tsx

```diff
@@ -45,12 +28,11 @@ function ProductShelf({
     bordered = ProductCard.props.bordered,
     showDiscountBadge = ProductCard.props.showDiscountBadge,
   } = {},
-  ...variables
+  products,
```

Comment:
What did we gain from moving the query upwards?

Reply:
You have a point. 
In one of the approaches I was trying to move the query hooks closer to the page and far from the UI component. 
But I think this doesn't make sense anymore and also adds more complexity. 
I reverted this approach approach [here](https://github.com/vtex/faststore/pull/1967/commits/6fd19fe74b09e14ebd12ce5ece4eaf16c195aed1).